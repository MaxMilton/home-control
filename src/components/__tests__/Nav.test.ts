import NavHtml from '../Nav.html';

describe('Nav component', () => {
  it('renders correctly', () => {
    expect.assertions(3);
    const spy = jest.spyOn(global.console, 'error');
    function wrapper() {
      const target = document.createElement('div');
      new NavHtml({ target });
      expect(target.innerHTML).toMatchSnapshot();
    }
    expect(wrapper).not.toThrow();
    expect(spy).not.toHaveBeenCalled();
  });
});
