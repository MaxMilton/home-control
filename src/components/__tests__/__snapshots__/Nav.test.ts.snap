// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders correctly 1`] = `
"<header class=\\"mb4\\"><nav class=\\"con dfc nav\\"><a href=\\"/\\" id=\\"logo\\">🏠🤖</a>

    
    <a href=\\".\\" class=\\"nav-item nav-item-active\\">Home</a>
    <a href=\\"admin\\" class=\\"nav-item\\">Admin</a></nav></header>"
`;
